pkgs/development/python-modules/pyobjc/default.nix

```diff
@@ -1,22 +1,476 @@
-  };
+  macSdkVersion = "10.12";
 
   meta = with lib; {
     description = "A bridge between the Python and Objective-C programming languages";
     license = licenses.mit;
-    maintainers = with maintainers; [ ];
+    maintainers = with maintainers; [ lukegb ];
     homepage = "https://pythonhosted.org/pyobjc/";
+    platforms = platforms.darwin;
+  };
+  src = stdenvNoCC.mkDerivation rec {
+    pname = "pyobjc-src";
+    inherit version meta;
+
+    src = fetchFromGitHub {
+      owner = "ronaldoussoren";
+      repo = "pyobjc";
+      rev = "af0349674b08126e52f2d055cc5c2d917afc3913";
+      sha256 = "1d58hrzwaalssdfg1018n2rg2k2270b30c7gb1w0nk6q3r3fccq2";
+    };
+
+    patches = [
+      # Read macOS SDK version from environment variable, rather than using system version.
+      ./macos-sdk-version.patch
+      ./pyobjc-setup-sdk-version.patch
+
+      # Fix LaunchServices/CoreServices.LaunchServices.
+      ./launch-services.patch
+
+      # We don't package the iTunesLibrary framework.
+      ./pyobjc-setup-drop-itunes.patch
+    ];
+
+    postPatch = ''
+      substituteInPlace pyobjc-core/Modules/objc/libffi_support.h \
+        --replace "ffi/ffi.h" "ffi.h"
+
+      for f in pyobjc-framework-*/pyobjc_setup.py; do
+        cp pyobjc-core/Tools/pyobjc_setup.py "$f"
+      done
+    '';
+
+    buildPhase = "true";
+    installPhase = ''
+      cp -R . $out
+    '';
+  };
+
+  subdir = name: attrs: let
+    frameworkInputs = lib.optionals (attrs ? frameworks) attrs.frameworks;
+    buildInputs = lib.optionals (attrs ? buildInputs) attrs.buildInputs;
+    deps = lib.optionals (attrs ? deps) attrs.deps;
+    skipCoreDep = if attrs ? skipCoreDep then attrs.skipCoreDep else false;
+    pythonModuleName = if name == "pyobjc-core" then "objc" else lib.removePrefix "pyobjc-framework-" name;
+  in buildPythonPackage {
+    pname = name;
+    inherit version src;
+
+    sourceRoot = "${src.name}/${name}";
+    propagatedBuildInputs = deps ++ lib.optional (!skipCoreDep) modules.pyobjc-core;
+    buildInputs = frameworkInputs ++ buildInputs ++ [ libobjc frameworks.Foundation ];
+
+    # Tests for -core don't run, and then the other tests don't work anyway.
+    doCheck = false;
+    pythonImportsCheck = [ pythonModuleName ];
+
+    NIX_MACOS_SDK_VERSION = macSdkVersion;
+
+    # Don't let them turn warnings into errors.
+    NIX_CFLAGS_COMPILE = "-Wno-error";
+
+    # Avoid `clang-7: error: argument unused during compilation: '-fno-strict-overflow' [-Werror,-Wunused-command-line-argument]` spam
+    hardeningDisable = ["strictoverflow"];
+
+    preCheck = ''
+      # Needed for PyObjCTest.test_bundleFunctions.TestBundleFunctions
+      export HOME=/var/empty
+    '';
+
+  };
+
+  moduleDeps = with frameworks; with modules; {
+    pyobjc-core = {
+      frameworks = [ Carbon Cocoa ];
+      buildInputs = [ libffi ];
+      deps = [ setuptools ];
+      skipCoreDep = true;
+    };
+
+    pyobjc-framework-AVFoundation = {
```

Comment:
I think I want to compare this list to the one I generated a while ago just to be sure https://github.com/NixOS/nixpkgs/pull/102306/files#diff-5e39f153e7ec38ba21422b8d82732044a0b0d899f463e5d3a9251cbdc38016e0R47.